app/code/Magento/ReviewGraphQl/Model/Resolver/Product/Reviews.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ReviewGraphQl\Model\Resolver\Product;
+
+use Magento\Catalog\Model\Product;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Review\Model\ResourceModel\Review\Collection;
+use Magento\ReviewGraphQl\Model\DataProvider\ProductReviewsDataProvider;
+
+/**
+ * Product reviews resolver, used by GraphQL endpoints to retrieve product's reviews
+ */
+class Reviews implements ResolverInterface
+{
+    /**
+     * @var ProductReviewsDataProvider
+     */
+    private $productReviewsDataProvider;
+
+    /**
+     * @param ProductReviewsDataProvider $productReviewsDataProvider
+     */
+    public function __construct(
+        ProductReviewsDataProvider $productReviewsDataProvider
+    ) {
+        $this->productReviewsDataProvider = $productReviewsDataProvider;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if ($args['currentPage'] < 1) {
+            throw new GraphQlInputException(__('currentPage value must be greater than 0.'));
+        }
+        if ($args['pageSize'] < 1) {
+            throw new GraphQlInputException(__('pageSize value must be greater than 0.'));
+        }
+
+        /** @var Product $product */
+        $product = $value['model'];
+        $args['productId'] = $product->getId();
+
+        /** @var Collection $reviewsCollection */
+        $reviewsCollection = $this->productReviewsDataProvider->getData($args);
```

Comment:
Please, do not pass the `args` array to the data provider. You may pass `productId`, `pageSize` and `currentPage` as separate arguments..

Reply:
Done ✅